VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -102,9 +102,14 @@ public void onDestroy() {
     public void onEventMainThread(DiscussionCommentPostedEvent event) {
         if (discussionThread.containsComment(event.getComment())) {
             discussionThread.incrementCommentCount();
-            courseDiscussionResponsesAdapter.setDiscussionThread(discussionThread);
+            if (event.getParent() == null) {
+                // We got a response
+                courseDiscussionResponsesAdapter.addNewResponse(event.getComment());
+            } else {
+                // We got a comment to a response
+                courseDiscussionResponsesAdapter.addNewComment(event.getParent());
+            }
```

Comment:
However this part of my comment still applies I guess:

> Note that to be technically correct, the response would also need to be validated as the child of the adapter's thread.

Since we're using posted events instead of Activity results, we should validate it. I guess this should be done at the Fragment level for all cases.